Edit profile

jacek15157 opened the pull request:
this is what I've done so far, look on this and tell me what should I change. Still don't know how it will work with redux and how to handle uploading photos and certifications. It's quazi responsvie.

inb4
Certifications and Birthday are independent component bc it was easier for me to style it. It was bc of Pop-up after click on data input filed.

jacek15157 committed:
added most of components to edit profile

jacek15157 committed:
clean up

jskurzynska commented on src/components/Navbar/Navbar.js:
```diff
@@ -85,7 +88,7 @@ class NavbarUnlogged extends Component {
                 </a>
                 <div className="dropdown-divider" />
-                <a className="dropdown-item" href="#">
+                <a className="dropdown-item" href="/edit-profile">
```
"/edit-profile" should be in constant variable

jskurzynska commented on src/components/PersonalData/Birthday/Birthday.js:
```diff
@@ -0,0 +1,21 @@
+const Birthday = props => {
+  return (
+    <div>
+      <label htmlFor="exampleInputName">Data Urodzin</label>
```
why do you need htmlFor?

jskurzynska commented on src/components/PersonalData/Birthday/Birthday.js:
```diff
@@ -0,0 +1,21 @@
+      <DatePicker
+        
+        id="birthday"
+        dateFormat="dd/MM/yyyy"
```
please put date format in variable

jskurzynska commented on src/components/PersonalData/Birthday/Birthday.js:
```diff
@@ -0,0 +1,21 @@
+        id="birthday"
+        dateFormat="dd/MM/yyyy"
+        selected={props.birthday}
+        onChange={props.birthdayHandler}
```
I think it would be better to call it *birthdayOnChange" or sth like that - it would be more intuitive

jskurzynska commented on src/components/PersonalData/Certyfications/Certyfications.js:
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const certyfications = () => {
```
I don't know how big it will be but for now I don't see the point in separated component for that

jskurzynska commented on src/components/PersonalData/PersonalData.js:
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import TextField from "./TextField/TextField.js";
+const personalData = props => {
```
Could you add empty like between imports and component? it would be more readable

jskurzynska commented on src/components/PersonalData/PersonalData.js:
```diff
@@ -0,0 +1,17 @@
+const personalData = props => {
+  return (
+    <TextField
+      email={props.email}
```
please use destructuring props

jskurzynska commented on src/components/PersonalData/TextField/TextField.css:
```diff
@@ -0,0 +1,3 @@
+.bc-color {
```
bc? what does it mean?

jskurzynska commented on src/components/PersonalData/TextField/TextField.js:
```diff
@@ -0,0 +1,40 @@
+  return (
+    <form>
+      <div className="form-group bc-color">
+        <label htmlFor="exampleInputName">Email</label>
```
htmlFor again - the same question - why?

jskurzynska commented on src/components/PersonalData/TextField/TextField.js:
```diff
@@ -0,0 +1,40 @@
+          className="form-control"
+          id="email"
+          value={props.email}
+          disabled={true}
```
you don't need true here

jskurzynska commented on src/components/PersonalData/TextField/TextField.js:
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import "./TextField.css";
+const TextField = props => {
```
"TextField" says nothing about the role of this component

jskurzynska commented on src/components/PersonalData/TextField/TextField.js:
```diff
@@ -0,0 +1,40 @@
+          className="form-control"
+          id="lastname"
+          defaultValue={props.lastName}
+          onChange={props.lastNameHandler}
```
please use destructuring props

jskurzynska commented on src/components/PersonalData/TextField/TextField.js:
```diff
@@ -0,0 +1,40 @@
+          value={props.email}
+          disabled={true}
+        />
+        <label htmlFor="exampleInputName">Imię</label>
```
could you put all texts in variables? it will be easier if we decide to add different language for app

jskurzynska commented on src/components/PersonalData/TextField/TextField.js:
```diff
@@ -0,0 +1,40 @@
+          className="form-control"
+          id="phone"
+          defaultValue={props.phone}
+          onChange={props.phoneHandler}
```
the same situation as birthdayHandler

jskurzynska commented on src/Auxiliary.js:
```diff
@@ -0,0 +1,3 @@
+const auxiliary = props => props.children;
+
+export default auxiliary;
```
I don't see the point in this

jskurzynska commented on src/containers/LoginFormContainer.js:
```diff
@@ -54,30 +54,35 @@ export default class LoginFormContainer extends Component {
-          this.props.history.push(HomePage);
-        else if(response.status === 201)
+      .then(response => {
+        if (response.status === 200) this.props.history.push(HomePage);
```
please use { }
I know that is not your code, but could change it?

jacek15157 committed:
fixed formatting

jacek15157 committed:
added buttons, descriptions, responsiveness

jacek15157 committed:
quick change to make basic user data look better

jacek15157 committed:
realtime validation, big redundancy need a repair

jacek15157 committed:
updated component did mount

jacek15157 committed:
upgrade quality of validation code

jacek15157 committed:
enable uploade picture

EnjoyTheNoise committed:
Merge remote-tracking branch 'origin/dev' into EditProfile

EnjoyTheNoise committed:
Refactored EditForm child components

fixed various jsx errors, rewritten into classes for standard React component lifecycle, deleted unnecessary components, moved files to designated locations, simplified props passing

EnjoyTheNoise committed:
Moved render logic to component

EnjoyTheNoise committed:
Refactored EditProfileContainer

Unified and simplified validation logic, organised state object, removed unnecessary properties, added email validation

EnjoyTheNoise committed:
Fix naming convention to match backend

EnjoyTheNoise committed:
Added feature to distinguish first login from the rest

EnjoyTheNoise committed:
Normalized Facebook/Google reducers initialState

EnjoyTheNoise committed:
Fixed edit link in navbar (was <a> html tag)

EnjoyTheNoise committed:
Added redux boilerplate for editing user profile

EnjoyTheNoise committed:
Added redux boilerplate for getting user if such one exists

EnjoyTheNoise committed:
registered new reducers in root

EnjoyTheNoise committed:
Fixed naming convention in edit form

EnjoyTheNoise committed:
Connected edit container with api through redux, fixed naming convention

Added different flow for first login - user will be added to db

EnjoyTheNoise reviewed (approved):
Applied my changes to edit form logic, connected it with api via redux + added missing parts of flow such as getting user and so on. Ready for review
